### PR TITLE
Mayuran/get in touch

### PR DESCRIPTION
**Pull request by mayuran-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/features/hooks/use-contact-form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import { useState } from 'react'
+import * as yup from 'yup'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useForm } from 'react-hook-form'
+import { validation_regex } from 'common/validation'
+import { trimObjectValues } from 'common/utility'
+import { contact_us_form_post_data } from 'common/constants'
+
+const error_message = {
+    mail: '_t_Email is required._t_',
```
Comment (hasan-deriv) [inline src/features/hooks/use-contact-form/index.tsx:10]: email: '_t_Email is required._t_',

File: src/features/hooks/use-contact-form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    mobile_number: yup.string().required(error_message['mobile_number']),
+})
+
+type FormDataType = yup.InferType<typeof schema>
+type FormStateType = {
+    is_loading_form: boolean
+    is_submitted: boolean
+    is_submission_fail: boolean
+}
+type FieldNameType = 'full_name' | 'mobile_number' | 'company_name' | 'email'
```
Comment (hasan-deriv) [inline src/features/hooks/use-contact-form/index.tsx:32]: type FieldNameType = keyof typeof error_message

File: src/features/pages/partners/deriv-prime/index.tsx [hunk 3]
```diff
@@ -12,6 +14,9 @@ const DerivPrime = () => {
                 title="_t_Liquidity solutions provider | Deriv Prime_t_"
             />
             <PaymentAgentAffiliateNav />
+            <Flex.Box padding_block={'40x'} margin_block="20x">
```
Comment (hasan-deriv) [inline src/features/pages/partners/deriv-prime/index.tsx:17]: I think, you don't need this wrapper anymore..

File: src/features/styles/theme/inputs/text.styles.scss [hunk 4]
```diff
@@ -31,13 +31,14 @@
                 outline: none;
                 ~ .text_input_label {
                     inset-block-start: -8px;
+                    color: $color-black;
```
Comment (hasan-deriv) [inline src/features/styles/theme/inputs/text.styles.scss:34]: This css will change the all inputs. Are we changing all?

File: src/features/pages/partners/deriv-prime/index.tsx (hunk 3 above)
Comment (mayuran-deriv) [reply src/features/pages/partners/deriv-prime/index.tsx:17]: that's to give a view in this page @aswathy-deriv  will remove this once merged

File: src/features/styles/theme/inputs/text.styles.scss (hunk 4 above)
Comment (mayuran-deriv) [reply src/features/styles/theme/inputs/text.styles.scss:34]: we are using this in only one place in home page its a single field, I thought every fields need to behave like same , only the label color is going to change so I changed this

File: src/features/hooks/use-contact-form/index.tsx (hunk 2 above)
Comment (mayuran-deriv) [reply src/features/hooks/use-contact-form/index.tsx:32]: sure

File: src/features/hooks/use-contact-form/index.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+}
+const schema = yup.object({
+    full_name: yup.string().required(error_message['full_name']),
+    company_name: yup.string().required(error_message['company_name']),
+    email: yup
+        .string()
+        .required(error_message['mail'])
+        .matches(validation_regex.email, { message: error_message['mail'] })
+        .email(error_message['mail']),
+    mobile_number: yup.string().required(error_message['mobile_number']),
```
Comment (hasan-deriv) [inline src/features/hooks/use-contact-form/index.tsx:23]: Are we not validating mobile number?

**Review by hasan-deriv — COMMENTED** [review]
After successful submission, form data should be cleared, right?

File: src/features/hooks/use-contact-form/index.tsx (hunk 5 above)
Comment (mayuran-deriv) [reply src/features/hooks/use-contact-form/index.tsx:23]: no man,  i asked unfortunately no

**Comment by mayuran-deriv** [comment]
@hasan-deriv no man we are not clearing I raised this issue too but they asked me to keep the data after submmision

**Review by hasan-deriv — APPROVED** [review]

**Review by aswathy-deriv — APPROVED** [review]

**Comment by habib-deriv** [comment]
The placeholder's last letter is visible

Screenshot 2023-08-24 at 11 47 38 AM

File: src/features/hooks/use-contact-form/index.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                        type: 'manual',
+                        message: error_message[fieldName],
+                    })
+                    has_validation_error = true
+                }
+            })
+
+            if (has_validation_error) {
+                setFormState({ ...form_state, is_loading_form: false })
+                return // Exit the function early
```
Comment (habib-deriv) [inline src/features/hooks/use-contact-form/index.tsx:65]: Can remove the comment
```suggestion
                return // Exit the function early
```

File: src/features/hooks/use-contact-form/index.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    const contact_us_form = useForm<FormDataType>({
+        mode: 'onChange',
+        resolver: yupResolver(schema),
+    })
+
+    const on_submit = async (data) => {
+        setFormState({ is_loading_form: true, is_submission_fail: false, is_submitted: false })
+        try {
+            const clean_data = trimObjectValues(data)
+            // Validation and error messages for each field if user filled data with only white space
```
Comment (habib-deriv) [inline src/features/hooks/use-contact-form/index.tsx:49]: Can remove this as well, the function name is handling the documentation
```suggestion
            
```

**Review by habib-deriv — APPROVED** [review]
LGTM

**Pull request merged**